File: Pancake Sorting.cpp
class Solution {
public:
    static vector<int> pancakeSort(vector<int> arr) {
        const int n = size(arr);
        vector<int> st;

        auto add = [&] (int x) {
            if (x > 1) {
                if (!st.empty() && x == st.back())
                    st.pop_back();
                else
                    st.push_back(x);
            }
        };

        auto dew = [&] (int j, int i) {
            if (i != j) {
                add(i + 1);
                add(i - j + 1);
                add(i - j);
                add(i - j - 1);
                add(i - j);
                add(i + 1);
            }
        };

        vector<int> indices(n);
        for (int i = 0; i < n; ++i)
            indices[arr[i] - 1] = i;
        for (int i = 0; i < n; ++i) {
            int pull = indices[i];
            dew(i, pull);
            swap(indices[arr[i] - 1], indices[i]);
            swap(arr[i], arr[pull]);
        }
        return st;
    }
};

File: pancake-sorting.py
class Solution:
    def pancakeSort(self, arr: List[int]) -> List[int]:
        length=len(arr)
        maxx=0
        out=[]
        
        for l in range (1,length):
          maxx= arr[:length].index(max(arr[:length]))
          out.append(maxx+1)
          j=0
          while j<=maxx:
              arr[j],arr[maxx]=arr[maxx],arr[j]
              j=j+1
              maxx=maxx-1  
          print(arr)
          arr[0:length]=arr[0:length][::-1]
          out.append(length)
          print(arr)
          length=length-1
        return out